### PR TITLE
SLING-10150 - sling:hideChildren incorrectly hiding parent of parent, when asterisk is used in combination with whitelisting

### PR DESCRIPTION
**Pull request by henrykuijpers**
Failing test case that shows the issue with sling:hideChildren and the incorrect handling of the whitelisting in combination with asterisk

### CODE REVIEW

File: src/main/java/org/apache/sling/resourcemerger/impl/MergingResourceProvider.java [hunk 1]
```diff
@@ -110,17 +110,14 @@ public ParentHidingHandler(final Resource parent, final boolean traverseParent)
...
-                            } else {
-                                onlyUnderlying = false;
-                            }
+                            final boolean onlyUnderlying = value.equals("*");
                             final ExcludeEntry entry = new ExcludeEntry(value, onlyUnderlying);
                             // check if this entry is applicable at all (always assuming the worst case, i.e. non local resource)
                             final Boolean hides = hides(entry, previousAncestorName, false);
-                            if (hides != null && hides.booleanValue() == true) {
-                                this.entries.add(new ExcludeEntry("*", entry.onlyUnderlying));
+                            if (hides != null) {
```
Comment (kwin) [inline src/main/java/org/apache/sling/resourcemerger/impl/MergingResourceProvider.java:117]: I am not sure TBH, although the documentation states 
> E.g. giving the values [!child1,*] hides all children except for the one with the name child1

I am wondering if that really makes sense, as then having multiple negated entries are not correctly considered, i.e. `[!child1,!child2,*]`. 
What we need is probably an AND logic over all values for negated values and an OR logic over all values for non-negated values. It gets complicated when mixing both though...

Comment (henrykuijpers) [reply src/main/java/org/apache/sling/resourcemerger/impl/MergingResourceProvider.java:117]: I will try to make a unit test for this. But I believe some additional discussion is going on now, also.

Comment (kwin) [reply src/main/java/org/apache/sling/resourcemerger/impl/MergingResourceProvider.java:117]: Please look at the alternative approach from https://github.com/apache/sling-org-apache-sling-resourcemerger/pull/4.

**Comment by henrykuijpers** [comment]
Closing this PR, as #4 has the better way of doing it

**Pull request closed without merging**